connectivity/check/context.go

```diff
@@ -43,7 +43,8 @@ type ConnectivityTest struct {
 	echoServices      map[string]Service
 	externalWorkloads map[string]ExternalWorkload
 
-	tests map[string]*Test
+	tests     []*Test
```

Comment:
nit: 

Creating a structure of this maybe a bit nicer to express they work together. 

```
struct TestSet {
  tests []*Test
  set    map[string]struct{}
}
```

Appropriate methods can be written.

Reply:
I'm not sure this is worth the additional code. There's only one place where we need to edit both at the same time and we're very unlikely to need to change the set in more places.

Reply:
fair enough.